### DISCUSSION TITLE
Is it possible to detect a compass click event?

### DISCUSSION THREAD

**Discussion by juanbrk** [mistic100/Photo-Sphere-Viewer#1436] [General]
I'm noticing that when rotating the viewer using compass navigation, the viewer's `pitch` gets reset to 0 (the default value, I think). I would like to maintain the previously set pitch to avoid losing context on this kind of rotation. 

https://github.com/user-attachments/assets/1e60e006-5f2f-40a0-b0d7-165106a41354

I can store the viewer' `textureX` and `textureY` provided by listening to the `PositionUpdatedEvent`  ([API Reference](https://photo-sphere-viewer.js.org/api/classes/Core.events.PositionUpdatedEvent.html)) or to the `BeforeRotateEvent` ([API reference](https://photo-sphere-viewer.js.org/api/classes/Core.events.BeforeRotateEvent.html)), but cannot discriminate between the rotation from navigation using the `compass` or any other kind of rotation. 

If I could detect that the `compass` has been clicked and the viewer has rotated by listening to any available events, I could simply restore the `viewer`'s pitch to Its previous position. 

Is there any simpler way to achieve this?

**Comment by mistic100** [comment]
Please do not use Discussions for bugs, feature and support !

I think the behaviour you want (keep current pitch) should be the default, please create a new issue about that.
